src/morphodict/lexicon/importer/test_importer.py

```diff
@@ -0,0 +1,530 @@
+    assert len(lemma_definitions) == 2
+
+    assert definitions_match(
+        lemma_definitions, [["A bear.", "MD"], ["bear, black bear", "CW"]]
+    )
+
+    target_kws = list(TargetLanguageKeyword.objects.all())
+    assert set(kw.text for kw in target_kws) == {"black", "bear"}
+    assert all(kw.wordform == lemma for kw in target_kws)
+
+    # Since everything here was fully analyzable, there shouldn’t be any extra
+    # indexing.
+    assert SourceLanguageKeyword.objects.count() == 0
+
+
+def debug(wf):
+    print(repr(wf))
+    for d in wf.definitions.all():
+        print(f"  {d.text} {d.source_ids}")
+
+
+@parametrize_reimport
+def test_import_lemma_auto_translation(db, reimport):
+    import_test_file("single-word.importjson", reimport=reimport)
+
+    debug(Wordform.objects.get(slug="maskwa"))
+
+    plural_wf = Wordform.objects.get(text="maskwak")
+    debug(plural_wf)
+    assert definitions_match(
+        plural_wf.definitions.all(),
+        [["bears.", "🤖MD"], ["bears, black bears", "🤖CW"]],
+    )
+
+    for d in plural_wf.definitions.all():
+        source = d.auto_translation_source
+        assert source is not None
+        assert d.citations.first().abbrv == "🤖" + source.citations.first().abbrv
+
+
+@parametrize_reimport
+@parametrize_translate_wordforms
+def test_import_lemma_auto_translation_with_multiple_sources(
+    db, translate_wordforms, reimport
+):
+    import_test_file(
+        "lemma-with-multiple-definition-sources.importjson",
+        translate_wordforms=translate_wordforms,
+        reimport=reimport,
+    )
+
+    wf = Wordform.objects.get(slug="amisk")
+    assert definitions_match(wf.definitions.all(), [["beaver", ["MD", "CW"]]])
+
+    if translate_wordforms:
+        plural_wf = Wordform.objects.get(text="amiskwak")
+        assert definitions_match(
+            plural_wf.definitions.all(), [["beavers", ["🤖MD", "🤖CW"]]]
+        )
+
+
+@parametrize_reimport
+class TestSourceLanguageKeywordPopulation:
+    @pytest.fixture(autouse=True)
+    def setup(self, db, settings, reimport):
+        settings.MORPHODICT_ENABLE_FST_LEMMA_SUPPORT = True
+
+        import_test_file(
+            "items-needing-source-language-keywords.importjson", reimport=reimport
+        )
+
+    def test_indexing_of_non_matching_slug(self):
+        """
+        When the slug does not match the text of an analyzable wordform, make
+        sure the slug gets indexed so that it is searchable.
+        """
+
+        wf = Wordform.objects.get(slug="amisk2")
+
+        kw = SourceLanguageKeyword.objects.get(wordform=wf)
+        assert kw.text == "amisk2"
+
+    def test_indexing_of_unanalyzable_form(self):
+        """
+        Usually the FST handles finding the lemma for a source-language search;
+        that doen’t work when the FST doesn’t know about a term.
+
+        ‘maskwa’ is of course analyzable but in the test data for this test, no
+        ‘analysis is given in the importjson.
```

Comment:
There's either an extra or a missing ' in this sentence. (not a huge deal by any means)

Reply:
Thanks—that comes from a bug in the IntelliJ plugin I use to wrap text at 80 chars.